doc/developer/design/20220516_transactional_consistency.md

```diff
@@ -0,0 +1,244 @@
+# Materialize Transactional Consistency
+
+## Summary
+
+A description of Materialize’s consistency guarantees with respect to the operations that occur inside of Materialize.
+Operations include: `SELECT`, `INSERT`, `UPDATE`, and `DELETE` statements (but not `TAIL`)
```

Comment:
What I think might be helpful is to mention that `TAIL` is somewhere in between serializable and strict serializable, in that you will see all writes in a total order, that total order will match the real time order that the writes took place in, but you may not see each write immediately (i.e. there is some delay after it commits and when it shows up in a `TAIL`). 

Does that sound reasonable?

Reply:
This is more than I meant to ask for, actually! All I really meant to ask for was that `TAIL` being a form of "read" may reveal certain orderings as invalid that otherwise we might not notice.

E.g. you do `w1` and then `w2`. In principle we could let `w2` happen at `t1` and `w1` happen at `t2 > t1`, if there isn't a read in between. We probably shouldn't, but we could if we only care about `SELECT` and `INSERT`. The presence of `TAIL` would reveal that this is unacceptable.